tests/test_logic.py

```diff
@@ -531,6 +532,41 @@ def test_Controller_on_sync_success(homedir, config, mocker):
     co.file_missing.emit.assert_called_once_with(missing.source.uuid, missing.uuid, str(missing))
 
 
+def test_Controller_on_sync_success_when_current_user_deleted(mocker, homedir):
+    co = Controller("http://localhost", mocker.MagicMock(), mocker.MagicMock(), homedir)
+
+    class DeletedUser(db.User):
+        @property
+        def username(self):
+            raise sqlalchemy.orm.exc.ObjectDeletedError(attributes.instance_state(db.User()), None)
+
+    co.authenticated_user = DeletedUser()
+    co.api = "not None"
+    co.gui = mocker.MagicMock()
+
+    co.on_sync_success()
+
+    assert co.authenticated_user is None
+    assert co.api is None
+    assert co.is_authenticated is False
```

Comment:
Maybe I'm missing something: isn't `co.authenticated_user is None` redundant if we're already asserting that `co.is_authenticated is False`?

Reply:
Since both properties exists on the Controller, the test makes sure that they are both what we expect them to be after the client session is closed. 

Not sure if you are suggesting a refactor here, but we could rewrite the Controller so that the boolean property (1) checks that the auth user exists and (2) checks that the auth user's token is valid, but I think you'd agree that we should be careful not to introduce unrelated changes into the bug fix.

Reply:
100% Let's not introduce unrelated changes in this PR!
The assertions caught my eye, maybe we can look into that part of the controller at some point (I'm not worried about it).